### ISSUE TITLE
variable explorer does not show variable for connected remote kernel

### ISSUE THREAD

**Issue by jontis** [spyder-ide/spyder#7850]
Variable explorer works fine for a local kernel.
Is this the way it's meant to work?
I love the variable explorer and it's a great strength that makes me come back to Spyder again and again.

If it's not a bug, please make it a feature request.

Spyder 3.3.1
Python 3.6.6 
IPython 6.5.0

**Comment by ccordoba12** [comment]
@jontis, the variable explorer is working now (i.e. after we released Spyder 3.3.0) in any kind of kernel: local (be it on the same environment Spyder is installed or on different envs) and remote (on any machine you have ssh access to and can install our little `spyder-kernels` package).

Please read this

https://docs.spyder-ide.org/ipythonconsole.html#connect-to-an-external-kernel

to see how to connect to any kernel.

**Comment by gsedej** [comment]
I installed the "Spyer 4" (Spyder 3.6), but "Variable explorer" does not work in my case.
Working case: 
as in documentation, we create kernel `python3 -m spyder_kernels.console` and connect to it in spyder. I can create variable, disconnet and reconnect, the variable is still visible.

Not working case:
Open `ipython3 qtconsole`, run some data-heavy reading (training images for neural network), get the kernel ID `from IPython.lib import kernel; kernel.get_connection_file()` connect to it's kernel in spyer. The input line is correct ˙In [85]` but "Variable explorer" is gray. I can interact with existing (external) ipython kernel via the integrated "ipython console". No error in console where `spyder3` was ran.

Ubuntu 18.04
Spyer via pip: `pip install --pre -U spyder --user`

![slika](https://user-images.githubusercontent.com/6927198/58476861-554e3180-8152-11e9-96d5-53ad15d89ecd.png)

**Comment by ccordoba12** [comment]
@gsedej, only our kernels (i.e. the ones provided by `spyder-kernels`) have the necessary additions for our Variable Explorer to interact with them.

**Comment by gsedej** [comment]
Thanks for reply.

**Comment by camilomarino** [comment]
I installed spyder-kernels on both computers, but the variable explorer does not work. I am using spyder 4, is it correct that this happens now?

I have Ubuntu 18.04 and spyder 4 on both PCs.

**Comment by BCArg** [comment]
The variable explorer is not working for me either, although it works on my local spyder session. 

I have successfully connected to the remote host according to the instructions on the [official documentation](http://docs.spyder-ide.org/ipythonconsole.html#connect-to-an-external-kernel).

I have launched the spyder kernel with `python -m spyder_kernels.console` on both remote and my local machine (though the connection works only if I launch this command on the remote host) and still the variables produced by my script are not showing up on the variable explorer.

Is this a bug or am I missing any step?
